cmd/prysmctl/withdrawal/withdrawal.go

```diff
@@ -0,0 +1,188 @@
+			}
+		}
+	}
+	return callWithdrawalEndpoint(ctx, beaconNodeHost, setWithdrawalAddressJsons)
+}
+
+func verifyWithdrawalCertainty(r io.Reader, request *apimiddleware.SignedBLSToExecutionChangeJson) error {
+	au := aurora.NewAurora(true)
+	withdrawalConfirmation := request.Message.ToExecutionAddress
+	fmt.Println(au.Red("===================================="))
+	fmt.Println("YOU ARE ATTEMPTING TO CHANGE THE BLS WITHDRAWAL(" + au.Red(request.Message.FromBLSPubkey).String() + ") ADDRESS " +
+		"TO AN ETHEREUM ADDRESS(" + au.Red(request.Message.ToExecutionAddress).String() + ") FOR VALIDATOR INDEX(" + au.Red(request.Message.ValidatorIndex).String() + "). ")
+	_, err := withdrawalPrompt(withdrawalConfirmation, r)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func callWithdrawalEndpoint(ctx context.Context, host string, request []*apimiddleware.SignedBLSToExecutionChangeJson) error {
+	body, err := json.Marshal(request)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal json")
+	}
+	fullpath := host + apiPath
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, fullpath, bytes.NewBuffer(body))
+	if err != nil {
+		return errors.Wrap(err, "invalid format, failed to create new Post Request Object")
+	}
+	req.Header.Set("Content-Type", "application/json")
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("API request to %s responded with a status other than OK - status %v, body %v", fullpath, resp.Status, resp.Body)
+	}
+	log.Infof("Successfully published messages to update %d withdrawal addresses.", len(request))
+
+	log.Info("retrieving list of withdrawal messages known to node...")
+	req, err = http.NewRequestWithContext(ctx, http.MethodGet, fullpath, nil)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	resp, err = client.Do(req)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("API request to %s responded with a status other than OK - status %v, body %v", fullpath, resp.Status, resp.Body)
+	}
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			log.WithError(err).Error("could not close response body")
+		}
+	}(resp.Body)
+	poolResponse := &apimiddleware.BLSToExecutionChangesPoolResponseJson{}
+	if err := json.NewDecoder(resp.Body).Decode(poolResponse); err != nil {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return errors.Wrap(err, "failed to read response body")
+		}
+		return errors.Wrap(err, fmt.Sprintf("invalid format, unable to read response body: %v", string(body)))
+	}
+	log.Infoln("known withdrawal messages to node, but not necessarily incorporated into any block yet: ")
+	for _, signedMessage := range poolResponse.Data {
+		log.Infof("validator index: %s with set withdrawal address: 0x%s", signedMessage.Message.ValidatorIndex, signedMessage.Message.ToExecutionAddress)
+	}
```

Comment:
I think instead of showing the pool status you need to check against each one of the messages requested to see if they are in the pool. The reason is that a user that is just trying to change a single message at the fork (like all of us small stakers will), may see hundred of thousands of messages here!

Reply:
ah didn't realize that will fix this.

Reply:
Let me know if this approach is too naive... or if you have other suggestions.